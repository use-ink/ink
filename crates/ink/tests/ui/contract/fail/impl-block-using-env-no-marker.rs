#![allow(unexpected_cfgs)]

#[ink::contract]
mod contract {
    #[ink(storage)]
    pub struct Contract {}

    impl Contract {
        #[ink(constructor)]
        pub fn constructor() -> Self {
            Self {}
        }
    }

    impl Contract {
        #[ink(message)]
        pub fn message(&self) {
            self.message_impl();
        }
    }

    impl Contract {
        fn message_impl(&self) {
            let _ = self.env().caller();
        }
    }
}

fn main() {}
