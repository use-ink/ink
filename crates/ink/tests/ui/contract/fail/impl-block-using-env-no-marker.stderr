error[E0599]: no method named `env` found for reference `&Contract` in the current scope
  --> tests/ui/contract/fail/impl-block-using-env-no-marker.rs:22:26
   |
22 |             let _ = self.env().caller();
   |                          ^^^ method not found in `&Contract`
   |
   = help: items from traits can only be used if the trait is in scope
help: trait `Env` which provides `env` is implemented but not in scope; perhaps you want to import it
   |
1  + use ink::codegen::Env;
   |
